server/asset/bch/live_test.go

```diff
@@ -0,0 +1,97 @@
+// +build bchlive
+//
+// go test -v -tags bchlive -run UTXOStats
+// -----------------------------------
+// Grab the most recent block and iterate it's outputs, taking account of
+// how many UTXOs are found, how many are of an unknown type, etc.
+//
+// go test -v -tags bchlive -run P2SHStats
+// -----------------------------------------
+// For each output in the last block, check it's previous outpoint to see if
+// it's a P2SH or P2WSH. If so, takes statistics on the script types, including
+// for the redeem script.
+//
+// go test -v -tags bchlive -run LiveFees
+// ------------------------------------------
+// Test that fees rates are parsed without error and that a few historical fee
+// rates are correct
+
+package bch
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/server/asset/btc"
+	"github.com/btcsuite/btcutil"
+)
+
+var (
+	bch *BCHBackend
+	ctx context.Context
+)
+
+func TestMain(m *testing.M) {
+	// Wrap everything for defers.
+	doIt := func() int {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithCancel(context.Background())
+		wg := new(sync.WaitGroup)
+		defer func() {
+			cancel()
+			wg.Wait()
```

Comment:
I noticed this panics if `Connect` does not succeed and thus wg.Add is never called.  We probably don't want to defer this closure unless Connect returns without error.